sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts

```diff
@@ -239,4 +252,79 @@ export class ApplicationService {
       })
       .getMany();
   }
+
+  /**
+   * Retrieves second disbursements eligible for a specific notification (likely disbursement
+   * date has passed and the institution has not yet approved the second COE/disbursement).
+   * This method applies several criteria to filter eligible disbursements:
+   * - There's a pending disbursement schedule
+   * - Application status is completed
+   * - No notification of this type (messageId 30) has been sent for this assessment
+   * - Disbursement date has passed today
+   * @returns An array of eligible disbursements with relevant details for notification.
+   */
+  async getSecondDisbursementsStillPending(): Promise<
+    SecondDisbursementStillPending[]
+  > {
+    // Sub query to defined if a notification was already sent to the current assessment.
+    const notificationExistsQuery = this.notificationRepo
+      .createQueryBuilder("notification")
+      .select("1")
+      .where("notification.notificationMessage.id = :messageId")
+      .andWhere(
+        "notification.metadata->>'assessmentId' = assessment.id :: text",
+      )
+      .getQuery();
+    // Retrieves disbursements with disbursement date has passed today and notification
+    // has not been sent for completed applications.
+    const disbursements = await this.disbursementScheduleRepo
+      .createQueryBuilder("disbursement")
+      .select("COUNT(assessment.id)", "assessmentCount")
+      .addSelect("MAX(disbursement.id)", "disbursementId")
+      .innerJoin("disbursement.studentAssessment", "assessment")
+      .innerJoin("assessment.application", "application")
+      .where(`NOT EXISTS (${notificationExistsQuery})`)
+      .andWhere("application.applicationStatus = :applicationStatusCompleted")
+      .andWhere("disbursement.disbursementDate < :today")
+      .groupBy("assessment.id")
+      .setParameters({
+        messageId:
+          NotificationMessageType.StudentSecondDisbursementNotification,
+        applicationStatusCompleted: ApplicationStatus.Completed,
+        today: getISODateOnlyString(new Date()),
```

Comment:
Minor, we do use Postgres `CURRENT_DATE` directly into the query sometimes, no need to provide it as a parameter.